DasharoPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c

```diff
@@ -156,73 +170,21 @@ PlatformRegisterFvBootOption (
                   &NewOption, BootOptions, BootOptionCount
                   );
 
-  if (OptionIndex == -1) {
+  if (OptionIndex == -1 && FileExists) {

```

Comment:
I think this lacks a check for `Remove` to prevent adding an option on removal:
```suggestion
  if (OptionIndex == -1 && FileExists && !Remove) {
```